ФЗ8

gennadiyniki opened the pull request:
ФЗ8

gennadiyniki committed:
ФЗ8

gennadiyniki committed:
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

VadimZharkov commented on src/tasks/Subtask.java:
```diff
@@ -1,20 +1,37 @@
-        return epicId;
+
+    @Override
+    public boolean hasIntersectionWith(Task other) {
```
Переопределять не нужно, если  в реализации метода ничего не изменилось. При вызове автоматом возьмется метод из базового класса.

VadimZharkov commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -53,25 +66,38 @@ public Epic addEpic(Epic epic) {
         }
-        epic.addSubtask(subtask.getId());
+
+        if (subtask.getStartTime() != null && subtask.getDuration() != null) {
```
Лучше здесь воспользоваться реализованным hasTimeIntersection для сокращения кода и единообразия с другими методами

VadimZharkov commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -276,4 +323,116 @@ protected void updateEpicStatus(Epic epic) {
+
+    public List<Task> getPrioritizedTasks() {
+        return prioritizedTasks.stream()
+                .sorted(Comparator.comparing(Task::getStartTime))
```
Можно не сортировать, т.к. в prioritizedTasks все уже лежит в отсортированнном виде

gennadiyniki replied:
понял, спасибо, не учел этот момент.

VadimZharkov commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -53,25 +66,38 @@ public Epic addEpic(Epic epic) {
     @Override
     public void clearSubtasks() {
         subtasks.clear();
+        prioritizedTasks.removeIf(task -> task instanceof Subtask);
```
На курсе не рекомендуют использовать instanceof т.к. он нарушает принципы ООП. Проще говоря лучше когда объект сам про себя все знает, т.е. имеет соответствующие методы. Поэтому тут лучше task.getType()

VadimZharkov commented on test/manager/TaskManagerTest.java:
```diff
@@ -0,0 +1,116 @@
+    protected abstract T createTaskManager();
+
+    @Test
+    public void test1GetEpicById() {
```
Название методов, полей или переменных с цифиркой test1... считается плохой практикой. Лучше цифры убрать.

gennadiyniki replied:
принял, убрал цифры.

VadimZharkov reviewed (changes requested):
Хорошая работа, можено улучшить. См. комменты выше.

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java
